db_test.go

```diff
@@ -935,6 +935,55 @@ func TestDB_Retention(t *testing.T) {
 	testutil.Equals(t, int64(100), db.blocks[0].meta.MaxTime) // To verify its the right block.
 }
 
+func TestSizeBasedRetention(t *testing.T) {
+	db, close := openTestDB(t, &Options{
+		BlockRanges: []int64{100},
+	})
+	defer close()
+
+	// Add some empty blocks.
+	blocks := []*BlockMeta{
+		{ULID: ulid.MustNew(100, nil), MinTime: 100, MaxTime: 200},
+		{ULID: ulid.MustNew(200, nil), MinTime: 200, MaxTime: 300},
+		{ULID: ulid.MustNew(300, nil), MinTime: 300, MaxTime: 400},
+		{ULID: ulid.MustNew(400, nil), MinTime: 400, MaxTime: 500},
+		{ULID: ulid.MustNew(500, nil), MinTime: 500, MaxTime: 600},
+	}
+	for _, m := range blocks {
+		createEmptyBlock(t, filepath.Join(db.Dir(), m.ULID.String()), m)
+	}
+
+	// Figure out how large an empty block is.
+	emptyBlockDir := filepath.Join(db.Dir(), blocks[0].ULID.String())
+	emptyBlock, err := OpenBlock(emptyBlockDir, db.chunkPool)
+	testutil.Ok(t, err)
+	db.opts.MaxBytes = 4 * emptyBlock.Size()
```

Comment:
can't we use something like `len(blocks) * emptyBlock.Size()` ? (the same for few other places below.) This would be more predictable if we decide to test with more blocks.

also a small comment of what is this line for would be helpful.